mmv1/products/vertexai/IndexEndpointDeployedIndex.yaml

```diff
@@ -0,0 +1,267 @@
+      - !ruby/object:Api::Type::Integer
+        name: 'minReplicaCount'
+        description: The minimum number of machine replicas this DeployedModel will be always deployed on. This value must be greater than or equal to 1.
+        immutable: true
+        required: true
+      - !ruby/object:Api::Type::Integer
+        name: 'maxReplicaCount'
+        description: The maximum number of replicas this DeployedModel may be deployed on when the traffic against it increases.
+        immutable: true
+      # - !ruby/object:Api::Type::Array
+      #   name: 'autoscalingMetricSpecs'
+      #   description: The metric specifications that overrides a resource utilization metric (CPU utilization, accelerator's duty cycle, and so on) target value (default to 60 if not set). At most one entry is allowed per metric.
+      #   immutable: true
+      #   item_type: !ruby/object:Api::Type::NestedObject
+      #     properties:
+      #       - !ruby/object:Api::Type::String
+      #         name: metricName
+      #         required: true
+      #         description: |
+      #           The resource metric name.
+      #       - !ruby/object:Api::Type::String
+      #         name: target
+      #         default_value: "60"
+      #         description: |
+      #           The target resource utilization in percentage (1% - 100%) for the given metric; once the real usage deviates from the target by a certain percentage, the machine replicas change. The default value is 60 (representing 60%) if not provided.
+  - !ruby/object:Api::Type::Boolean
+    name: enableAccessLogging
+    description: If true, private endpoint's access logs are sent to Cloud Logging.
+    default_value: false
+  - !ruby/object:Api::Type::NestedObject
+    name: deploymentIndexAuthConfig
+    description: If set, the authentication is enabled for the private endpoint.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: authProvider
+        description: Defines the authentication provider that the DeployedIndex uses.
+        properties:
+          - !ruby/object:Api::Type::Array
+            name: audiences
+            output: true
+            description: The list of JWT audiences. that are allowed to access. A JWT containing any of these audiences will be accepted.
+            item_type: Api::Type::String
+          - !ruby/object:Api::Type::Array
+            name: allowedIssuers
+            output: true
+            description: |
+              A list of allowed JWT issuers. Each entry must be a valid Google service account, in the following format: service-account-name@project-id.iam.gserviceaccount.com
+            item_type: Api::Type::String
+  - !ruby/object:Api::Type::Array
+    name: reservedIpRanges
+    description: A list of reserved ip ranges under the VPC network that can be used for this DeployedIndex.
+    item_type: Api::Type::String
+    ignore_read: true
```

Comment:
If this is `ignore_read: true` due to [this reason here](https://github.com/GoogleCloudPlatform/magic-modules/pull/11039#issuecomment-2209316648):

>    * when importing (read) the plan returns to be nil if not set in config, yet the resp from the api returns an empty array. The empty array makes sense in this case since the `reservedIpRanges` field is a list type. I would've expected it to handle it as empty rather than nil. Probably some extra logic for this.

Could the diff between null (from config) and an empty list (from API) be suppressed by a diff suppress func?

Or another way to think about it, what would a user do if they want to import a deployed index that has reserved ip ranges and reference those values elsewhere?

Reply:
DiffSuppressFuncs tend to work only with `string-like` attributes. Though I'm sure there's a work around that can suppress an empty list being set as null in state. We may have to do something similar like what was done here: https://github.com/GoogleCloudPlatform/magic-modules/pull/10936#issuecomment-2168954728

Reply:
Or maybe you could have a custom flattener that converts an empty array for that field to null, and otherwise doesn't change the value?

Reply:
had the same idea, trying it out right now.

Reply:
i may have just been importing with the wrong resource. Works fine now.